src/wagtail_vector_index/index/base.py

```diff
@@ -70,9 +95,45 @@ def query(
             merged_context,
             query,
         ]
+
         response = self.chat_backend.chat(user_messages=user_messages)
         return QueryResponse(response=response.text(), sources=sources)
 
+    async def aquery(self, query: str) -> AsyncQueryResponse[VectorIndexableType]:
+        """
+        Async version of the query method.
+        """
+        if not self.chat_backend.can_stream():
```

Comment:
Note: I've added this here as I think there will likely be an issue with developers expecting streaming support when using async, especially since we're abstracting from the LLM package, so theres no obvious way for them to know without tracing errors back.

I don't think this is the correct solution if it is a problem, but have added it here to highlight the potential issue and get some thoughts?